### DISCUSSION TITLE
Form layout not able to store data with belongsTo relation

### DISCUSSION THREAD

**Discussion by rmitesh** [filamentphp/filament#9230] [Help]
I have 2 models `User` and `Restaurant`.

Where the user has many restaurant relations concepts.

Models\User.php

```php
public function restaurants(): HasMany
{
    return $this->hasMany(Restaurant::class);
}
```

Models\Restaurant.php
```php

// Columns
// 'user_id', 'name',

public function user(): BelongsTo
{
    return $this->belongsTo(User::class);
}
```

Now, I have a form in the `RestaurantResource` where I have used 

```php
Forms\Components\Fieldset::make('Owner Details')
    ->relationship('user')
    ->schema([
      // Inputs
    ]),
    
Forms\Components\Fieldset::make('Restaurant Details')
    ->schema([
        // Inputs
    ])

```

but I am getting this error.

![image](https://github.com/filamentphp/filament/assets/48554454/3edebccd-ce12-42ab-9778-8f56bf7ed357)

I have also reviewed the official documentation https://filamentphp.com/docs/3.x/forms/advanced#saving-data-to-relationships but I feel I have followed the correct steps.

**Comment by leandrocfe** [comment]
Shouldn't be

```php
Forms\Components\Select::make('user_id')
    ->label('Owner')
    ->relationship('user', 'name')
```
?

**Reply by rmitesh** [reply]
So, first I need to create User/Owner then need to create Restaurant?
Or of course I will have to change the flow!

**Reply by leandrocfe** [reply]
Yes. You can also create new options in a modal using this

https://filamentphp.com/docs/3.x/forms/fields/select/#creating-a-new-option-in-a-modal
